mlx-nn/src/activation.rs

```diff
@@ -265,11 +265,11 @@ pub fn hard_swish(x: impl AsRef<Array>) -> Result<Array, Exception> {
 ///
 /// This splits the `axis` dimension of the input into two halves
 /// (`a` and `b`) and applies `a * sigmoid(b)`.
-#[derive(Debug, Clone, Default, ModuleParameters)]
-#[option_builder]
+#[derive(Debug, Clone, Default, ModuleParameters, GenerateBuilder)]
 pub struct Glu {
     /// The axis to split the input tensor. Default to [`Glu::DEFAULT_AXIS`] if not provided.
-    pub axis: Option<i32>,
+    #[optional(default_value = Glu::DEFAULT_AXIS)]
```

Comment:
do we support `default_value` for non optionals?  Otherwise it might be simpler for us to just keep the `Optional<T>` type and just annotate with `#[default_value()]`